tests/formats/xtc.cpp

```diff
@@ -79,3 +80,317 @@ TEST_CASE("Read files in XTC format") {
+        CHECK(approx_eq(cell.c(), 22, 1e-4));
+        CHECK(approx_eq(cell.alpha(), 33.333, 1e-4));
+        CHECK(approx_eq(cell.beta(), 44.444, 1e-4));
+        CHECK(approx_eq(cell.gamma(), 55.555, 1e-4));
+
+        frame = file.read();
+
+        CHECK(frame.step() == 200);
+        CHECK(approx_eq(frame.get("time")->as_double(), 20));
+        CHECK(approx_eq(frame.get("xtc_precision")->as_double(), 1000)); // default precision
+        CHECK(frame.size() == 10);
+
+        positions = frame.positions();
+        CHECK(approx_eq(positions[0], Vector3D(7, 8, 9), 1e-4));
+        CHECK(approx_eq(positions[2], Vector3D(4, 5, 6), 1e-4));
+
+        cell = frame.cell();
+        CHECK(cell.shape() == UnitCell::ORTHORHOMBIC);
+        CHECK(approx_eq(cell.a(), 30, 1e-4));
+        CHECK(approx_eq(cell.b(), 31, 1e-4));
+        CHECK(approx_eq(cell.c(), 32, 1e-4));
+    }
+
+    SECTION("Append on new trajectory") {
+        auto tmpfile = NamedTempPath(".xtc");
+
+        auto frame = Frame(UnitCell(10.111, 11.222, 12.333));
+        frame.set("time", 19.376);
+        frame.set("xtc_precision", 10000); // higher precision for more decimal places
+        frame.add_atom(Atom("A"), {1.999, 2.888, 3.777});
+        frame.add_atom(Atom("B"), {4, 5, 6});
+        frame.add_atom(Atom("C"), {7, 8, 9});
+        frame.add_atom(Atom("D"), {10, 11, 12});
+        frame.add_atom(Atom("E"), {13, 14, 15});
+        frame.add_atom(Atom("F"), {16, 17, 18});
+        frame.add_atom(Atom("G"), {19, 20, 21});
+        frame.add_atom(Atom("H"), {22, 23, 24});
+        frame.add_atom(Atom("I"), {25, 26, 27});
+        frame.add_atom(Atom("J"), {28, 29, 30});
+
+        auto file = Trajectory(tmpfile, 'a');
+        file.write(frame);
+        CHECK(file.nsteps() == 1);
+        file.close();
+
+        // now read every thing back and check
+        file = Trajectory(tmpfile, 'r');
+
+        frame = file.read();
+
+        CHECK(frame.step() == 0); // default step
+        CHECK(approx_eq(frame.get("time")->as_double(), 19.376, 1e-4));
+        CHECK(approx_eq(frame.get("xtc_precision")->as_double(), 10000));
+        CHECK(frame.size() == 10);
+
+        auto positions = frame.positions();
+        CHECK(approx_eq(positions[0], Vector3D(1.999, 2.888, 3.777), 1e-4));
+        CHECK(approx_eq(positions[2], Vector3D(7, 8, 9), 1e-4));
+
+        auto cell = frame.cell();
+        CHECK(cell.shape() == UnitCell::ORTHORHOMBIC);
+        CHECK(approx_eq(cell.a(), 10.111, 1e-4));
+        CHECK(approx_eq(cell.b(), 11.222, 1e-4));
+        CHECK(approx_eq(cell.c(), 12.333, 1e-4));
+    }
+}
+
+TEST_CASE("Check Errors") {
+    auto tmpfile = NamedTempPath(".xtc");
+    auto file = Trajectory(tmpfile, 'w');
+
+    auto frame = Frame();
+    frame.add_atom(Atom("A"), {1, 2, 3});
+    file.write(frame);
+
+    frame = Frame();
+    frame.add_atom(Atom("A"), {1, 2, 3});
+    frame.add_atom(Atom("B"), {4, 5, 6});
+    CHECK_THROWS_AS(file.write(frame), FormatError);
```

Comment:
Can you replace this by `CHECK_THROWS_WITH` and check that the error message is as expected?

Reply:
Didn't know about that :)